sdks/java/extensions/ml/src/main/java/org/apache/beam/sdk/extensions/ml/RecommendationAIImportCatalogItems.java

```diff
@@ -0,0 +1,197 @@
+import org.apache.beam.sdk.transforms.ParDo;
+import org.apache.beam.sdk.values.KV;
+import org.apache.beam.sdk.values.PCollection;
+import org.apache.beam.sdk.values.PCollectionTuple;
+import org.apache.beam.sdk.values.TupleTag;
+import org.apache.beam.sdk.values.TupleTagList;
+import org.checkerframework.checker.nullness.qual.Nullable;
+import org.joda.time.Duration;
+import org.json.JSONObject;
+
+/**
+ * A {@link PTransform} connecting to the Recommendations AI API
+ * (https://cloud.google.com/recommendations) and creating {@link CatalogItem}s. *
+ *
+ * <p>Batch size defines how many items are created at once per batch (max: 5000).
+ *
+ * <p>The transform consumes {@link KV} of {@link String} and {@link GenericJson}s (assumed to be
+ * the catalog item id as key and contents as value) and outputs a PCollectionTuple which will
+ * contain the successfully created and failed catalog items.
+ *
+ * <p>It is possible to provide a catalog name to which you want to add the catalog item (defaults
+ * to "default_catalog").
+ */
+@AutoValue
+@SuppressWarnings({"nullness"})
+public abstract class RecommendationAIImportCatalogItems
+    extends PTransform<PCollection<KV<String, GenericJson>>, PCollectionTuple> {
+
+  public static final TupleTag<CatalogItem> SUCCESS_TAG = new TupleTag<CatalogItem>() {};
+  public static final TupleTag<CatalogItem> FAILURE_TAG = new TupleTag<CatalogItem>() {};
+
+  public static Builder newBuilder() {
+    return new AutoValue_RecommendationAIImportCatalogItems.Builder();
+  }
+
+  public abstract Builder toBuilder();
+
+  /** @return ID of Google Cloud project to be used for creating catalog items. */
+  public abstract @Nullable String projectId();
+
+  /** @return Name of the catalog where the catalog items will be created. */
+  public abstract @Nullable String catalogName();
+
+  /** @return Size of input elements batch to be sent in one request. */
+  public abstract Integer batchSize();
+
+  /**
+   * @return Time limit (in processing time) on how long an incomplete batch of elements is allowed
+   *     to be buffered.
+   */
+  public abstract Duration maxBufferingDuration();
+
+  public RecommendationAIImportCatalogItems withProjectId(String projectId) {
+    return this.toBuilder().setProjectId(projectId).build();
+  }
+
+  public RecommendationAIImportCatalogItems withCatalogName(String catalogName) {
+    return this.toBuilder().setCatalogName(catalogName).build();
+  }
+
+  public RecommendationAIImportCatalogItems withBatchSize(Integer batchSize) {
+    return this.toBuilder().setBatchSize(batchSize).build();
+  }
+
+  /**
+   * The transform converts the contents of input PCollection into {@link CatalogItem}s and then
+   * calls the Recommendation AI service to create the catalog item.
+   *
+   * @param input input PCollection
+   * @return PCollection after transformations
+   */
+  @Override
+  public PCollectionTuple expand(PCollection<KV<String, GenericJson>> input) {
```

Comment:
consider returning a ImportResult instead of PCollectionTuple (see like this for BQ.Write: https://github.com/apache/beam/blob/master/sdks/java/io/google-cloud-platform/src/main/java/org/apache/beam/sdk/io/gcp/bigquery/WriteResult.java#L37)